### PR TITLE
Fix time active

### PR DESCRIPTION
**Pull request by Justin-Koningsberger**
https://trello.com/c/VOPnHzf4/48-workease-suspend-of-toch-niet-suspend

The `@state[part][:activity_start]` was reset to zero, and time active was calculated
 as `Time.now.to_i - @state[part][:activity_start]`, giving a very high number.

### CODE REVIEW

File: workease.rb [hunk 1]
```diff
@@ -102,6 +103,8 @@ def check_device(keyboard_id, mouse_id)
   def check(part)
     @semaphore.synchronize do
       time = Time.now.to_i
+      next if time < @pause_until
```
Comment (lucasluitjes) [inline workease.rb:106]: Why was this added? Some sort of optimization?

File: workease.rb [hunk 2]
```diff
@@ -224,14 +227,14 @@ def rest_timer(time, activity)
     pid = Process.fork do
       sleep time
       `paplay ./sounds/service-login.ogg`
-      `xmessage #{message} -center -timeout 3`
+      `xmessage #{message} -center -timeout 2`
```
Comment (lucasluitjes) [inline workease.rb:231]: Why was this changed?

File: workease.rb [hunk 3]
```diff
@@ -244,7 +247,7 @@ def warn(reason)
       @warn_log << message
     elsif Time.now.to_i > @pause_until
       `paplay ./sounds/when.ogg`
-      @pause_until = Time.now.to_i + 5
+      @pause_until = Time.now.to_i + 2
```
Comment (lucasluitjes) [inline workease.rb:251]: Why was this changed?

**Review by lucasluitjes — CHANGES_REQUESTED** [review]
Mostly looks good, but I saw some unrelated changes?

Also: there is no test for this bug. While your explanation makes it clear what the problem was and that it likely will be fixed now, it's always good practice to first add a test that reproduces the bug before you fix it. Which should fail when you run it, then after fixing the bug, the test should pass. This both helps you to make sure that you actually fixed the bug and also (maybe more importantly) means it will never come back again. AKA it prevents regressions.

File: workease.rb (hunk 1 above)
Comment (Justin-Koningsberger) [reply workease.rb:106]: This stops the check method from running while a warning is displayed. So you can click close, without that click generating another warning.

File: workease.rb (hunk 2 above)
Comment (Justin-Koningsberger) [reply workease.rb:231]: 3 seconds felt a little long for a message saying; 'all good, you can get back to work now'. Maybe it's just me, but when I was testing, it felt slow.

File: workease.rb (hunk 3 above)
Comment (Justin-Koningsberger) [reply workease.rb:251]: This should be 4 seconds, the warn method sleeps for 1 second, and displays the message for 3 seconds. If @pause_until is set to 4 seconds, you can click close on the message without it firing another warning straight away.

**Comment by Justin-Koningsberger** [comment]
I added a test for this bug, what do you think of the timing changes I commented on above? If you want I/we can revert those changes when resolving merge conflicts later.

**Review by lucasluitjes — APPROVED** [review]

**Comment by lucasluitjes** [comment]
All your comments make sense to me, approved.
